### PR TITLE
team mates can see same unit info as player owner

### LINKED ISSUE #282: Allow allies/team mates to see all details of units same as home nation sees
Comes in real handy with team based games.

### PR DESCRIPTION
**Pull request by jwrober**
Closes #282

### CODE REVIEW

**Comment by lmoureaux** [comment]
Middle click doesn't seem to be using the newly available information.

**Comment by jwrober** [comment]
Yea I def dorked that up. I'll look to push another commit later this week.

File: server/unittools.cpp [hunk 1]
```diff
@@ -2836,6 +2836,11 @@ void send_unit_info(struct conn_list *dest, struct unit *punit)
       if (pdata != nullptr) {
         BV_SET(pdata->can_see_unit, player_index(pplayer));
       }
+    } else if (pplayer->team == powner->team) {
```
Comment (lmoureaux) [inline server/unittools.cpp:2839]: To facilitate maintenance I would prefer this (from `player.h`):
```suggestion
    } else if (players_on_same_team(pplayer, powner)) {
```
Could we merge this with the previous `if`? After all, a player is always on its own team...

**Comment by lmoureaux** [comment]
Can you rebase for RTD?

**Review by lmoureaux — APPROVED** [review]
Let's merge this

**Pull request merged**